docs/onboarding/custom-framework-overview.md

```diff
@@ -0,0 +1,93 @@
+
+**Update** function takes a message from the component, updates the (immutable) state, and sends a message back. Can also return a promise (optional, but allows you to avoid a race state.)
+
+**Init** takes some params, returns a promise with a state.
+
+**View** takes props (optional arguments) in components, props must contain a state.
+
+#### `/grunt-configs/`
+
+GRUNT: calls the typescript compiler through grunt. The intent was a unified approach to front-end and back-end.
+
+#### `/src/front-end/typescript/index.tsx`
+
+- a single point of entry
+
+#### `/src/migrations`
+
+Purpose is to migrate the (mutable) schema and adhere to the type safe design principle. Custom migration framework to leverage type safety, using knex. Used to snapshot the data. The data, sometimes represented as ENUMs in migration scripts.
+
+#### `/src/scripts`
+
+more for development, not deployment. for example bcrypt for hashing passwords. a place to keep things for posterity.
+
+## Back-End
+
+ADT (algebraic data types) are used quite a bit; often used in functional programming and have been around for years. Each value constructor has a tag and a value. Example is color.
+
+ ``` typescript
+ type Color
+   = ADT<'red'>
+   | ADT<'blue'>
+   | ADT<'green'>
+   | ADT<'rgb', [number, number, number]>;
+
+ const red: Color = { tag: 'red', value: undefined };
+ const rgb: Color = { tag: 'rgb', value: [123, 255, 7] };
+ ```
+
+### `/src/backend/index.ts`
+
+This is the point of entry, a tonne of imports. calls the `start()` function which starts the app, catches all errors, validates environment variables, connects to the database, sets up the router (router is a custom layer on top of express - for, you guessed it, the purpose of type safety). Also has a `start()` function. One adapter `ExpressAdapter` ('back-end/lib/server/adapters') acts as a driver to make sure the routes are handled properly, manage sessions, hosts, port, max file size - the arguments here are the minimum number of arguments to get the router going. Namespaced for clarity, example is /admin.
+
+### `/src/backend/lib/crud.js`
+
+one server that does everything, all in one monolith solution
+
+### `/src/back-end/start.js`
+
+Keeps things consistent. A way to make sure that module imports are brought in (via moduleAlias).
+
+## Front-End
+
+Need to produce a single javascript file; typescript and browserify allow us to do this.
+
+#### `/src/front-end/typescript/index.ts`
+
+This is the entry point. Grunt takes care of a lot of the work. Similar to the back-end there is a `start()` function. Environment variables are set at build time, not at runtime. You get a state manager at start up. Listeners for window resizing.
+
+#### `/src/front-end/typescript/lib/app/index.tsx`
+
+*AppComponent* comes from the framework (`front-end/lib/framework`). init, update, view, router
```

Comment:
@bdolor Nice job on this file. Do you mind if add a few things directly in this branch?

Reply:
do I mind? do I mind? no, I do not mind (at all - and thanks!)